src/main/flight/interpolated_setpoint.c

```diff
@@ -58,9 +69,42 @@ FAST_CODE_NOINLINE float interpolatedSpApply(int axis, bool newRcFrame, ffInterp
         const float rxInterval = currentRxRefreshRate * 1e-6f;
         const float rxRate = 1.0f / rxInterval;
 
-        const float setpointSpeed = (rawSetpoint - prevRawSetpoint[axis]) * rxRate;
-        const float setpointAcceleration = (setpointSpeed - prevSetpointSpeed[axis]) * pidGetDT();
+        float setpointSpeed = (rawSetpoint - prevRawSetpoint[axis]) * rxRate;
+        float setpointAcceleration = setpointSpeed - prevSetpointSpeed[axis];
+        const float holdSteps = 2.0f;
+
+        // Glitch reduction code for identical packets
+        if (setpointSpeed == 0 && fabsf(rawSetpoint) < 0.98f * ffMaxRate[axis]) {
+            // identical packets, not at full deflection
+            if (holdCount[axis] < holdSteps && fabsf(rawSetpoint) > 2.0f && !bigStep[axis]) {
+                // holding the entire previous speed is best for missed packets, but bad for early packets
+                setpointSpeed = prevSetpointSpeed[axis] + prevAcceleration[axis];
+                setpointAcceleration = prevAcceleration[axis];
+                holdCount[axis] += 1.0f;
+            // hit timeout, or at centre and not moving, or prev big step, lock speed and acceleration to zero
+            } else {
+                holdCount[axis] -= 1.0f;
+                setpointSpeed = 0.0f;
+                setpointAcceleration = 0.0f;
```

Comment:
I don't undertand this?  To even get here setpointSpeed is already 0, so I think that's unneeded.  But why do we set acceleration to 0?  Alternative would be leaving it set to whatever line 73 calculated it to be?  Is 0 better?

Reply:
Yes, setting setpoint speed to 0 is not needed in 87, thanks. 
Acceleration depends on prevSetpointSpeed[axis]; this won't be zero if projected forward like  when holdCount was 2.  It does need to be forced to zero.